generator/README.md

```diff
@@ -0,0 +1,824 @@
+ * Required, NSString *
+ */
+@property (strong, nonatomic) NSString *appID;
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+
+```objc
+//  SDLGetCloudAppProperties.m
+//
+
+#import "SDLGetCloudAppProperties.h"
+#import "NSMutableDictionary+Store.h"
+#import "SDLRPCFunctionNames.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SDLGetCloudAppProperties
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+- (instancetype)init {
+    self = [super initWithName:SDLRPCFunctionNameGetCloudAppProperties];
+    if (!self) { return nil; }
+
+    return self;
+}
+#pragma clang diagnostic pop
+
+- (void)setAppID:(NSString *)appID {
+    [self.parameters sdl_setObject:appID forName:SDLRPCParameterNameAppId];
+}
+
+- (NSString *)appID {
+    NSError *error = nil;
+    return [self.parameters sdl_objectForName:SDLRPCParameterNameAppId ofClass:NSString.class error:&error];
+}
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+##### Response Functions (SDLRPCResponse)
+
+##### Response class with extra argument example (Alert Response):
+
+```xml
+<function name="Alert" functionID="AlertID" messagetype="response" since="1.0">     
+	<param name="success" type="Boolean" platform="documentation" mandatory="true">
+		<description> true if successful; false, if failed </description>
+	</param>
+
+	<param name="resultCode" type="Result" platform="documentation" mandatory="true">
+		<description>See Result</description>
+		<element name="SUCCESS"/>
+		<element name="INVALID_DATA"/>
+		<element name="OUT_OF_MEMORY"/>
+		<element name="TOO_MANY_PENDING_REQUESTS"/>
+		<element name="APPLICATION_NOT_REGISTERED"/>
+		<element name="GENERIC_ERROR"/>
+		<element name="REJECTED"/>
+		<element name="ABORTED"/>
+		<element name="DISALLOWED"/>
+		<element name="USER_DISALLOWED"/>
+		<element name="UNSUPPORTED_RESOURCE"/>
+		<element name="WARNINGS"/>
+	</param>
+
+	<param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
+		<description>Provides additional human readable info regarding the result.</description>
+	</param>
+
+	<param name="tryAgainTime" type="Integer" minvalue="0" maxvalue="2000000000" mandatory="false" since="2.0">
+		<description>
+			Amount of time (in seconds) that an app must wait before resending an alert.
+			If provided, another system event or overlay currently has a higher priority than this alert.
+			An app must not send an alert without waiting at least the amount of time dictated.
+		</description>
+	</param>
+</function>
+```
+
+As we can see it in the XML it requires an additional param `tryAgainTime` which is declared and implemented in the `SDLAlertResponse` class as follows:
+
+```objc
+//  SDLAlertResponse.h
+//
+
+#import "SDLRPCResponse.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * @since SDL 1.0.0
+ */
+@interface SDLAlertResponse : SDLRPCResponse
+
+/**
+ * @param @(tryAgainTime)
+ * @return A SDLAlertResponse object
+ */
+- (instancetype)initWithTryAgainTime:(UInt32)tryAgainTime;
```

Comment:
I know this is complicated, but I think that `init`s including optional parameters (like `tryAgainTime` is) should use the `NSNumber` version of the parameter and make it `nullable`. Mandatory parameters can use `UInt32` since they're not optional anyway. Would that be difficult to implement?

Reply:
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/88a2ae6d3ba1d92c3444893a7509078f5da1ac5e
```objc
- (instancetype)initWithTryAgainTime:(nullable NSNumber<SDLUInt> *)tryAgainTime;
```